### DISCUSSION TITLE
Failed To Initialize front interface issue

### DISCUSSION THREAD

**Discussion by sabinathapa375** [AltraMayor/gatekeeper#704] [Q&A]
This is the log of gatekeeper

root@gtn-Lenovo-G40-70:/var/log/gatekeeper# cat gatekeeper_2024_09_13_16_20.log 
Main/0: cycles/second = 1696073207, cycles/millisecond = 1696073, picosec/cycle = 589
Main/0: config: /usr/share/lua/5.1/gatekeeper/staticlib.lua:547: Failed to initilialize front interface
Main/0: Failed to configure Gatekeeper

Bus Info

root@gtn-Lenovo-G40-70:/var/log/gatekeeper# lshw -c network -businfo 

Bus info          Device      Class          Description
========================================================
pci@0000:01:00.0  enp1s0      network        RTL8111/8168/8411 PCI Express Gigabit Ethernet Controller
pci@0000:02:00.0              network        RTL8723BE PCIe Wireless Network Adapter

net.lua 

local log_level = staticlib.c.RTE_LOG_DEBUG
	local user = "gatekeeper"
 
	local front_ports = {"enp1s0"}
	local front_ips  = {"192.168.180.64"}
	local front_bonding_mode = staticlib.c.BONDING_MODE_8023AD
	local front_ipv4_vlan_tag = 0
	local front_ipv6_vlan_tag = front_ipv4_vlan_tag
	local front_vlan_insert = false
	local front_mtu = 1500
 
	local back_ports = {"enp1s0"}
	local back_ips  = {"192.168.180.64"}
	local back_bonding_mode = staticlib.c.BONDING_MODE_ROUND_ROBIN
	local back_ipv4_vlan_tag = 0x456
	local back_ipv6_vlan_tag = back_ipv4_vlan_tag
	local back_vlan_insert = true
	local back_mtu = 1500

**Comment by AltraMayor** [comment]
DPDK does not have drivers for Realtek NICs; see [DPDK's supported NICs](https://core.dpdk.org/supported/nics/). You may want to check the page [Hardware Requirements](https://github.com/AltraMayor/gatekeeper/wiki/Hardware-Requirements) from the wiki.

Besides the issue above, I see two other problems in your `net.lua`:
1. Variables `front_ports` and `back_ports` cannot share ports;
2. Variable `back_mtu` must be greater than `front_mtu` so no encapsulated packets to the Grantor servers are dropped.
